test/shared_utilities/data_handling.jl

```diff
@@ -0,0 +1,142 @@
+#=
+    Unit tests for ClimaLand DataHandling module
+=#
+
+using Dates
+using Test
+
+import ClimaCore: Fields
+import ClimaLand
+import ClimaLand: DataHandling
+using NCDatasets
+
+include(joinpath(pkgdir(ClimaLand), "test", "artifacts", "artifacts.jl"))
+temporal_data_path = era5_t2m_sp_u10n_dataset_path()
+static_data_path = era5_t2m_sp_u10n_static_dataset_path()
+
+@testset "DataHandler, TempestRegridder, static" begin
+    PATH = static_data_path
+    varname = "sp"
+    for FT in (Float32, Float64)
+        target_space =
+            ClimaLand.Domains.SphericalShell(;
+                radius = FT(6731e3),
+                depth = FT(1.0),
+                nelements = (4, 4),
+                npolynomial = 4,
+            ).space.surface
+
+        data_handler = DataHandling.DataHandler(PATH, varname, target_space)
+
+        @test DataHandling.available_dates(data_handler) == DateTime[]
+
+        field = DataHandling.regridded_snapshot(data_handler)
+        # Most basic testing, to get a sense that we are doing things right
+        @test field isa Fields.Field
+        @test axes(field) == target_space
+        @test eltype(field) == FT
+        @test minimum(field) >=
+              minimum(data_handler.file_reader.dataset[varname][:, :])
+        @test maximum(field) <=
+              maximum(data_handler.file_reader.dataset[varname][:, :])
+
+        close(data_handler)
+    end
+end
+
+@testset "DataHandler, TempestRegridder, time data" begin
+    for FT in (Float32, Float64)
+        target_space =
+            ClimaLand.Domains.SphericalShell(;
+                radius = FT(6731e3),
+                depth = FT(1.0),
+                nelements = (4, 4),
+                npolynomial = 4,
+            ).space.surface
+        PATH = temporal_data_path
+        varname = "sp"
```

Comment:
these could be moved outside of the for loop for consistency with the previous test (but it doesn't really matter)